pkg/deploy/predeploy.go

```diff
@@ -0,0 +1,268 @@
+	}
+
+	d.log.Infof("predeploying to %s", d.config.ResourceGroupName)
+	return d.deployments.CreateOrUpdateAndWait(ctx, d.config.ResourceGroupName, deploymentName, mgmtresources.Deployment{
+		Properties: &mgmtresources.DeploymentProperties{
+			Template:   template,
+			Mode:       mgmtresources.Incremental,
+			Parameters: parameters.Parameters,
+		},
+	})
+}
+
+func (d *deployer) configureServiceKV(ctx context.Context) error {
+	serviceKeyVaultURI := "https://" + d.config.Configuration.KeyvaultPrefix + "-svc.vault.azure.net/"
+
+	err := d.ensureEncryptionSecret(ctx, serviceKeyVaultURI)
+	if err != nil {
+		return err
+	}
+
+	err = d.ensureMonitoringCertificates(ctx, serviceKeyVaultURI)
+	if err != nil {
+		return err
+	}
+
+	return d.ensureServiceCertificates(ctx, serviceKeyVaultURI)
+}
+
+func (d *deployer) ensureEncryptionSecret(ctx context.Context, serviceKeyVaultURI string) error {
+	secrets, err := d.keyvault.GetSecrets(ctx, serviceKeyVaultURI, nil)
+	if err != nil {
+		return err
+	}
+
+	for _, secret := range secrets {
+		if filepath.Base(*secret.ID) == env.EncryptionSecretName {
+			return nil
+		}
+	}
+
+	key := make([]byte, 32)
+	_, err = rand.Read(key)
+	if err != nil {
+		return err
+	}
+
+	_, err = d.keyvault.SetSecret(ctx, serviceKeyVaultURI, env.EncryptionSecretName, keyvault.SecretSetParameters{
+		Value: to.StringPtr(string(key)),
+	})
+	return err
+}
+
+func (d *deployer) ensureMonitoringCertificates(ctx context.Context, serviceKeyVaultURI string) error {
+	for _, certificateName := range []string{
+		env.ClusterLoggingSecretName,
+		env.RPLoggingSecretName,
+		env.RPMonitoringSecretName,
+	} {
+		bundle, err := d.keyvault.GetSecret(ctx, d.config.Configuration.GlobalMonitoringKeyVaultURI, certificateName, "")
+		if err != nil {
+			return err
+		}
+
+		_, err = d.keyvault.ImportCertificate(ctx, serviceKeyVaultURI, certificateName, keyvault.CertificateImportParameters{
+			Base64EncodedCertificate: bundle.Value,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (d *deployer) ensureServiceCertificates(ctx context.Context, serviceKeyVaultURI string) error {
+	_, err := d.keyvault.SetCertificateIssuer(ctx, serviceKeyVaultURI, "OneCert", keyvault.CertificateIssuerSetParameters{
+		Provider: to.StringPtr("OneCert"),
+	})
+	if err != nil {
+		return err
+	}
+
+	certs := []struct {
+		certificateName string
+		commonName      string
+		eku             utilkeyvault.Eku
+	}{
+		{
+			certificateName: env.RPFirstPartySecretName,
+			commonName:      d.config.Location + "." + d.config.Configuration.RPParentDomainName,
+			eku:             utilkeyvault.EkuClientAuth,
+		},
+		{
+			certificateName: env.RPServerSecretName,
+			commonName:      d.config.Configuration.RPServerCertCommonName,
+			eku:             utilkeyvault.EkuServerAuth,
+		},
+	}
+
+	for _, c := range certs {
+		err = d.keyvault.CreateSignedCertificate(ctx, serviceKeyVaultURI, utilkeyvault.IssuerOnecert, c.certificateName, c.commonName, c.eku)
```

Comment:
```) keyvault.BaseClient#CreateCertificate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="BadParameter" Message="Request to manage certificate is rejected with reason: Bad Request" 
exit status 1
```
When we enable oneCert, dont we need to wait for approval?

Reply:
No the approval for OneCert is per the user/principal requesting the cert be generated not per keyvault. You can put the issuer on any keyvault but only an pre-authorized user can generate a cert.